core/src/main/java/org/mule/util/StringMessageUtils.java

```diff
@@ -255,7 +255,11 @@ public static String headersToString(MuleMessage m)
                     {
                         value = "<<<MuleMessage>>>";
                     }
-                    if (name.equals("password") || name.toString().contains("secret") || name.equals("pass") || name.toString().contains("authorization"))
+                    if (name.equals("password") || name.equals("pass"))
+                    {
+                        value = "****";
+                    }
+                    else if (name.toString().contains("authorization") || name.toString().contains("secret"))
```

Comment:
You can do the toString in the else and move the if inside for a better approach